src/framework/preferences/js/SyllabificationEnactor.js

```diff
@@ -218,29 +215,64 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
         var hyphenator = fluid.getGlobalValue(globalPath);
 
         // If the pattern file has already been loaded, return the hyphenator.
+        // This could happen if the pattern file is statically linked to the page.
         if (hyphenator) {
             promise.resolve(hyphenator);
             return promise;
         }
 
         var src = fluid.stringTemplate(pattern, that.options.terms);
+
         var injectPromise = that.injectScript(src);
         injectPromise.then(function () {
             hyphenator = fluid.getGlobalValue(globalPath);
             promise.resolve(hyphenator);
-        }, function (jqXHR, textStatus, errorThrown) {
-            that.events.onError.fire(
-                "The pattern file \"" + src + "\" could not be loaded.",
-                jqXHR,
-                textStatus,
-                errorThrown
-            );
-            // If the pattern file could not be loaded resolve the promise with out a hyphenator (undefined).
+        }, function () {
+            var errorMessage = fluid.stringTemplate(that.options.strings.patternLoadError, {src: src});
+            // Pass along arguments fired for the rejection, but with our errorMessage as the first argument
+            var args = [errorMessage].concat(fluid.values(arguments));
+            that.events.onError.fire.apply(null, args);
+
+            //TODO: A promise rejection would be more appropriate. However, we need to know when all of the hyphenators
+            //      have attempted to load and apply syllabification. The current promise utility,
+            //      fluid.promise.sequence, will reject the whole sequence if a promise is rejected, and prevent us from
+            //      knowing if all of the hyphenators have been attempted. We should be able to improve this
+            //      implementation once https://issues.fluidproject.org/browse/FLUID-5938 has been resolved.
+            //
+            // If the pattern file could not be loaded, resolve the promise without a hyphenator (undefined).
             promise.resolve();
         });
+
         return promise;
     };
 
+    /**
+     * Assembles an Object containing the information for locating the pattern file. If a pattern for the specific
+     * requested language code cannot be located, it will attempt to locate a fall back, by looking for a pattern
+     * supporting the generic language code. If no pattern can be found, `undefined` is returned as the `src` value.
+     *
+     *
+     * @param {String} lang - a valid BCP 47 language code. (NOTE: supported lang codes are defined in the
+     *                        `patterns`) option.
+     * @param {Object} patterns - an object mapping language codes to file paths for the pattern files. For example:
```

Comment:
Please refine the types a bit here - e.g. {Object.<String, String>} and we could have a small @typedef for the "pattern" object value type in the returned structure
https://jsdoc.app/tags-type.html

Reply:
@amb26 I've made the change to the comment regarding the types. Please let me know if I have addressed it appropriately.